src/VisualStudio/Xaml/Impl/Diagnostics/Analyzers/XamlDocumentDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Package = Microsoft.VisualStudio.Shell.Package;
+
+namespace Microsoft.CodeAnalysis.Xaml.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(StringConstants.XamlLanguageName)]
+    internal class XamlDocumentDiagnosticAnalyzer : DocumentDiagnosticAnalyzer
+    {
+        private IXamlDocumentAnalyzerService analyzerService;
+
+        private IXamlDocumentAnalyzerService AnalyzerService
+        {
+            get
+            {
+                if (this.analyzerService == null)
+                {
+                    IComponentModel componentModel = (IComponentModel)Package.GetGlobalService(typeof(SComponentModel));
+                    if (componentModel == null)
+                    {
+                        Debug.Fail("Missing IComponentModel!");
+                        return null;
+                    }
+
+                    this.analyzerService = componentModel.GetExtensions<IXamlDocumentAnalyzerService>().FirstOrDefault();
+                    Debug.Assert(this.analyzerService != null, "Missing IXamlDocumentAnalyzerService!");
+                }
+
+                return this.analyzerService;
+            }
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get
+            {
+                return this.AnalyzerService?.SupportedDiagnostics ?? ImmutableArray.Create<DiagnosticDescriptor>();
```

Comment:
Nit: Use `ImmutableArray<DiagnosticDescriptor>.Empty` to avoid allocating an empty backing array.